### ISSUE TITLE
tokenServer on subscribe triggered twice

### ISSUE THREAD

**Issue by MauricioKruijer** [azuqua/node-token-sockjs#9]
Hi there,
When a client subscribes to a channel, my backend seems to receive the event twice.
Can you help me find out what I'm doing wrong?
# server.js
```javascript
var path = require('path');
var express = require("express"),
http = require("http"),
redis = require("redis"),
sockjs = require("sockjs"),
TokenSocketServer = require("node-token-sockjs");

var redisClient = redis.createClient(),
    pubsubClient = redis.createClient();

var app = express(),
    socketServer = sockjs.createServer();

app.use(express.static(path.join(__dirname, 'public')));

var server = http.createServer(app);

var socketOptions = {
  prefix: "/sockets"
};
socketServer.installHandlers(server, socketOptions);
server.listen(3000, '0.0.0.0');

var tokenServer = new TokenSocketServer(app, redisClient, socketServer, {
  prefix: socketOptions.prefix,
  tokenRoute: "/socket/token",
  pubsubClient: pubsubClient,
  authentication: function(req, callback){
    var auth = {
      username:"Username",
      passhash: "serverjs"
    };
    return callback(null, auth);
  },
  debug: app.get("env") !== "production"//,
});
tokenServer.on("authentication", function(socket, auth, callback){
  tokenServer.rpc(socket, "pong", {from: "authentication", pong: new Date()});
  callback(null, auth);
});

tokenServer.on("subscribe", function(socket, data, callback){
  // this will be triggered twice for each client
  console.log("Socket attempting to subscribe: ", socket.auth, data.channel);
  callback(null, true);
});
```
# public/index.html
```html
<!DOCTYPE html>
<html>
<head lang="en">
  <meta charset="UTF-8">
  <title>SockJS Node</title>
  <!--<script src="javascripts/sockjs.js"></script> &lt;!&ndash; version 0.3.4 &ndash;&gt;-->
  <!--<script src="javascripts/tokensockjs.js"></script>-->
  <script src="https://raw.githubusercontent.com/sockjs/sockjs-client/master/dist/sockjs-0.3.4.js"></script>
  <script src="https://raw.githubusercontent.com/azuqua/token-sockjs-client/master/clients/browser/tokensockjs.js"></script>
  <script>
    console.log('Options');
    var options = {
      host: "localhost:3000",
      tokenPath: "/socket/token",
      socketPrefix: "/sockets",
      reconnect: true,
      authentication: {
        authuser: "aaa",
        pin: "x0x8uldkj"
      }
    };
    try {
      var actions = {
        pong: function(data, next) {
          console.log("pong received!", data);
          next(null, true);
        }
      };
      var socket = new TokenSocket(options, actions);
      socket.ready(function(error) {
        if(error) throw Error("Error creating websocket");
        console.log("connection made");
        socket.subscribe("channelJS");

        socket.onmessage(function(channel, message) {

        });
      });
    } catch(e){
      console.log("error");
      console.log(e);
    };
    console.log('socket');
  </script>
</head>
<body>
<h1>SockJS Test</h1>
</body>
</html>
```

So when I run `node server.js` and run the client I see the `Socket attempting to subscribe` message twice..

node v0.12.0
express 4.12.3

**Comment by aembke** [comment]
Good find, this was a subtle one. It should be fixed in the newest version of the client (2.0.8)

**Comment by MauricioKruijer** [comment]
Thanks :) Sorry for opening the issue in the wrong repo.. That's probably why I couldn't find it haha
